base/operators.jl

```diff
@@ -328,8 +328,15 @@ eltype(::Type{Any}) = Any
 eltype(t::DataType) = eltype(supertype(t))
 eltype(x) = eltype(typeof(x))
 
-# function pipelining
+# function pipelining, composition & negation
+
 |>(x, f) = f(x)
+∘(f, g) = (x...)->f(g(x...))
+
+!(f::Function) = (x...)->!f(x...)
+^(f::Function, g::Function) = (x...)->f(x...) ^ g(x...)
+|(f::Function, g::Function) = (x...)->f(x...) || g(x...)
```

Comment:
So maybe better use `&` and `|` for clarity. As you said, predicates are often quite cheap anyway. The distinction between `&` and `&&` is already subtle enough for non-professional programmers (i.e. most scientists) that we shouldn't increase confusion.